pgmpy/tests/test_models/test_BayesianNetwork.py

```diff
@@ -1018,7 +1019,12 @@ def test_predict(self):
         titanic.add_edges_from([("Sex", "Survived"), ("Pclass", "Survived")])
         titanic.fit(self.titanic_data2[500:])
 
-        p1 = titanic.predict(self.titanic_data2[["Sex", "Pclass"]][:30])
+        p1_variable_elimination = titanic.predict(
+            self.titanic_data2[["Sex", "Pclass"]][:30]
+        )
+        p1_belief_propagation = titanic.predict(
+            self.titanic_data2[["Sex", "Pclass"]][:30], algo=BeliefPropagation
+        )
```

Comment:
Could you add a test for approx inference as well? The values would be different, and you would need to test it using `assertAlmostEqual`.

Reply:
@ankurankan The test is failing for some reason. 
`p2_approx_inference = titanic.predict( self.titanic_data2[["Survived", "Pclass"]][:30], algo=ApproxInference )`
is showing the type error for not being a subclass of Inference.
`TypeError: Algorithm should be a valid pgmpy inference method. Got <class 'type'> instead.`

Reply:
As you can see here: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/inference/ApproxInference.py, ApproxInference is not a subclass of Inference, so you would need to handle it separately.

Reply:
@ankurankan Any other cases to be accounted for now? Or only ApproxInference, VariableElim, and BeliefProp?

Reply:
No

Reply:
The .predict() method [calls for map_query()](https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/models/BayesianNetwork.py#L676) when stochastic is False (default case) but there is no .map_query() for ApproxInference. So should this combination of parameters be disallowed or something else?

Reply:
You can add a map_query method to ApproxInference. The map_query is similar to the query method. The difference is that it returns the state with the highest probability instead of returning the probability distribution. For the map_query method, you would need to call the query method and then return the variables' states with the highest probability.